flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -264,21 +266,21 @@ public void attachJobGraph(List<AbstractJobVertex> topologiallySorted) throws Jo
 			LOG.debug(String.format("Attaching %d topologically sorted vertices to existing job graph with %d "
```

Comment:
We have to check that the parallelism of both vertices are the same, because the runtime results are already partitioned. I think a good error msg is the best solution at this point.